pkg/integration/tests/file/stage_renamed_range_select.go

```diff
@@ -0,0 +1,43 @@
+package file
+
+import (
+	"github.com/jesseduffield/lazygit/pkg/config"
+	. "github.com/jesseduffield/lazygit/pkg/integration/components"
+)
+
+var StageRenamedRangeSelect = NewIntegrationTest(NewIntegrationTestArgs{
```

Comment:
The use of renames here feels unnecessarily complicated, as it doesn't actually relate to the changes we've made, even if it was the situation that led you to make this PR. I would simplify this to just having four files:
* foo
* foobar
* baz/file
* bazbam/file

And trying to select all of them and staging them. That makes clearer the intent what we're defending against.